owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerLocator.java

```diff
@@ -1,9 +1,81 @@
 package edu.rice.owltorrent.network;
 
-import edu.rice.owltorrent.common.entity.Peer;
+import com.dampcake.bencode.BencodeInputStream;
+import com.google.common.io.ByteStreams;
 import edu.rice.owltorrent.common.entity.Torrent;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
 import java.util.List;
+import lombok.NonNull;
 
-public interface PeerLocator {
-  List<Peer> locatePeers(Torrent torrent);
+/**
+ * Tracker connector to fetch peers.
+ *
+ * @author bhaveshshah
+ */
+public class PeerLocator {
+
+  public static final String infoHash =
+      "%32%31%0b%4d%b8%4d%e5%c0%23%bf%cb%9f%40%64%8d%8c%9e%7c%a1%6e"; // TODO: retrieve from torrent
+  public static final String peerID = "owltorrentclientpeer";
+  public static final String port = "6991";
+  public static final String left = "0";
+  public static final String downloaded = "0";
+  public static final String uploaded = "0";
+  public static final String compact = "1";
+
+  public List<InetSocketAddress> locatePeers(@NonNull Torrent torrent) {
+    List<InetSocketAddress> addresses = new ArrayList<>();
+
+    try {
+      // Build request
+      StringBuilder builder = new StringBuilder("https://torrent.ubuntu.com/announce?");
+      builder.append("info_hash=" + infoHash + "&");
+      builder.append("peer_id=" + peerID + "&");
+      builder.append("port=" + port + "&");
+      builder.append("left=" + left + "&");
+      builder.append("downloaded=" + downloaded + "&");
+      builder.append("uploaded=" + uploaded + "&");
+      builder.append("compact=" + compact);
```

Comment:
Nit: Since these string concats are not in a loop, you could just use plus sign to concat as the compiler should be able to optimize that.

Reply:
Changed to plus sign concatenation in revision.